m2cgen/interpreters/mixins.py

```diff
@@ -124,15 +128,33 @@ def process_subroutine_queue(self, top_code_generator):
         subroutine queue.
         """
         self._subroutine_idx = 0
+        subroutines = []
 
-        while len(self.subroutine_expr_queue):
+        while self.subroutine_expr_queue:
             self._reset_reused_expr_cache()
             subroutine = self.subroutine_expr_queue.pop(0)
             subroutine_code = self._process_subroutine(subroutine)
+            subroutines.append((subroutine, subroutine_code))
```

Comment:
If my assumption in the previous comment is correct I don't think we actually need `subroutine` here and having only `subroutine_code` will suffice.